### ISSUE TITLE
Require types to be declared at the top of the file (after import)

### ISSUE THREAD

**Issue by gajus** [gajus/eslint-plugin-flowtype#168]
**Please describe what the rule should do:**

Require that all `type` declarations appear at the top of the file, first thing after any import declarations.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Good:

```js
import createAsyncTracker from '../createAsyncTracker';
import type {
  AdType
} from '../types';

type Unit = {|
  prefix: string,
  ignoreSuffix?: Array<string>,
  size: string,
  id: string,
  max?: number
|};

```

Bad:

```js
import createAsyncTracker from '../createAsyncTracker';
import type {
  AdType
} from '../types';

const foo = 'bar';

type Unit = {|
  prefix: string,
  ignoreSuffix?: Array<string>,
  size: string,
  id: string,
  max?: number
|};

```

**Comment by gajus** [comment]
@danharper Might be a good idea to create an issue template, like ESLint does:

* http://eslint.org/docs/developer-guide/contributing/new-rules
* https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md

Do you want me to create one?

Is the above information sufficient? (This issue)

**Comment by danharper** [comment]
👍  on an issue template
